lib/models/environment-variables.coffee

```diff
@@ -143,3 +143,126 @@ exports.remove = (id, callback) ->
 ####
 exports.isSystemVariable = (variable) ->
 	return /^RESIN_/.test(variable.name)
+
+exports.device = {}
+
+###*
+# device.getAll callback
+# @callback module:resin.models.environment-variables.device~getAllCallback
+# @param {(Error|null)} error - error
+# @param {EnvironmentVariable[]} environmentVariables - environment variables
+###
+
+###*
+# @summary Get all device environment variables
+# @public
+# @function
+#
+# @param {(String|Number)} deviceId - device id
+# @param {module:resin.models.environment-variables.device~getAllCallback} callback - callback
+#
+# @example
+#	resin.models.environmentVariables.device.getAll 51, (error, environmentVariables) ->
+#		throw error if error?
+#		console.log(environmentVariables)
+###
+exports.device.getAll = (deviceId, callback) ->
+	return pine.get
+		resource: 'device_environment_variable'
+		options:
+			filter:
+				device: deviceId
+			expand: 'device'
+			orderby: 'env_var_name asc'
+	.tap (environmentVariables) ->
+		if _.isEmpty(environmentVariables)
+			throw new errors.ResinNotAny('device environment variables')
```

Comment:
I guess is debatable, but I'll keep the current behaviour since all the other SDK "getAll" functions behave the same way, but definitely something to consider in the future.